posthog-core/src/index.ts

```diff
@@ -696,7 +743,11 @@ export abstract class PostHogCore extends PostHogCoreStateless {
     }
   }
 
-  getSessionId(): string | undefined {
+  getSessionId(): string {
+    if (!this._isInitialized) {
+      return ''
```

Comment:
Probably better to return undefined?
otherwise we need to add a check here https://github.com/PostHog/posthog-js-lite/blob/2d1da92df028f2915e5147c9fab07f77fd094f60/posthog-core/src/index.ts#L695
an empty string would trigger an ingestion warning I guess? it does with a few other important flags at least such as anon_id.

Reply:
Not sure tbh. I think the issue here is that if we make it undefined then the caller _always_ has to check for undefined. Is it nicer to force the user to always make that check or to just tolerate that it might be undefined.

Internal calls to that will never be undefined as they are always called from wrapped functions. Not sure what is more annoying for the user.

Reply:
It's ok that it returns an empty (and invalid string) if written in the method docs, I also prefer that over `undefined`.

Reply:
I'll add some comments